### ISSUE TITLE
Avellaneda_MM - Hanging orders not recreated when max order age is reached

### ISSUE THREAD

**Issue by rxlxrxsx** [CoinAlpha/hummingbot#3610]
**Describe the bug**
There are instances that when the hanging order hit the max_order_age, it was never recreated. When max order age is reached and the hanging orders were canceled, two normal orders were created. Additional info from @phbrgnomo:
It seems that the bot is recognizing these two new normal orders created right after the hanging order cancelation, as both normal AND hanging:

Normal because they use the price/order size as a normal order
Hanging because these orders are not canceled after order_refresh_time (which I set to 300s). They are canceled only after the max_order_age had passed. Only after the max_order_age pass for these conflicting orders, the hanging orders are recreated, along with the two other normal orders.

<img width="1207" alt="Screen Shot 2021-06-20 at 3 44 17 PM" src="https://user-images.githubusercontent.com/64377055/122666404-8be6c100-d1df-11eb-8c90-f202be523f97.png">

**Steps To Reproduce**
1. Run a avellaneda_market_making bot with hanging order enabled
2. For testing purposes you can set the max order age to a lower value

**Release version**
dev-0.41.0

**Attachments**
[ave_hanging.zip](https://github.com/CoinAlpha/hummingbot/files/6681902/ave_hanging.zip)

**Comment by rapcmia** [comment]
Adding another instance of this issue on hanging orders and max_order_age:
While looking into it, I noticed that the hanging order will only be recreated after the next order cycle. In my example, I set max_order_age to 3:07s (187) and waited for the even to recreate the hanging order. It was not recreated and the bot waited for my order_refresh_time (60s) then it creates both active and hanging orders again.


https://user-images.githubusercontent.com/73840223/122667849-a1f87f80-d1e7-11eb-9882-d52966e8acc3.mp4

```
# the bot created bid/ask order
2021-06-20 16:35:16,135 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Created LIMIT_MAKER BUY order x-XEKWYICX-BFTBC1624178116034452 for 16.00000000 FRONT-BTC.
2021-06-20 16:35:16,135 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178116.0, "type": "OrderType.LIMIT_MAKER", "trading_pair": "FRONT-BTC", "amount": "16.00000000", "price": "0.00001963", "order_id": "x-XEKWYICX-BFTBC1624178116034452", "exchange_order_id": "19328846", "leverage": 1, "position": "NILL", "event_name": "BuyOrderCreatedEvent", "event_source": "binance"}
2021-06-20 16:35:16,645 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Created LIMIT_MAKER SELL order x-XEKWYICX-SFTBC1624178116034453 for 15.90000000 FRONT-BTC.
2021-06-20 16:35:16,645 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178116.0, "type": "OrderType.LIMIT_MAKER", "trading_pair": "FRONT-BTC", "amount": "15.90000000", "price": "0.00001968", "order_id": "x-XEKWYICX-SFTBC1624178116034453", "exchange_order_id": "19328845", "leverage": 1, "position": "NILL", "event_name": "SellOrderCreatedEvent", "event_source": "binance"}

# the hanging order reached max_order_age and cancelled
2021-06-20 16:35:20,028 - 17028 - hummingbot.strategy.hanging_orders_tracker - INFO - Reached max_order_age=187.0sec hanging order: HangingOrder(order_id='x-XEKWYICX-BFTBC1624177932997306', trading_pair='FRONT-BTC', is_buy=True, price=Decimal('0.00001961'), amount=Decimal('12.80000000')). Renewing...
2021-06-20 16:35:20,028 - 17028 - hummingbot.strategy.avellaneda_market_making.avellaneda_market_making - INFO - (FRONT-BTC) Cancelling the limit order x-XEKWYICX-BFTBC1624177932997306. [clock=2021-06-20 08:35:20+00:00]
2021-06-20 16:35:20,176 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Successfully cancelled order x-XEKWYICX-BFTBC1624177932997306.
2021-06-20 16:35:20,176 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178120.0, "order_id": "x-XEKWYICX-BFTBC1624177932997306", "exchange_order_id": null, "event_name": "OrderCancelledEvent", "event_source": "binance"}
2021-06-20 16:36:15,997 - 17028 - hummingbot.strategy.avellaneda_market_making.avellaneda_market_making - INFO - (FRONT-BTC) Cancelling the limit order x-XEKWYICX-BFTBC1624178116034452. [clock=2021-06-20 08:36:16+00:00]
2021-06-20 16:36:15,997 - 17028 - hummingbot.strategy.avellaneda_market_making.avellaneda_market_making - INFO - (FRONT-BTC) Cancelling the limit order x-XEKWYICX-SFTBC1624178116034453. [clock=2021-06-20 08:36:16+00:00]
2021-06-20 16:36:16,091 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Successfully cancelled order x-XEKWYICX-BFTBC1624178116034452.
2021-06-20 16:36:16,091 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178176.0, "order_id": "x-XEKWYICX-BFTBC1624178116034452", "exchange_order_id": null, "event_name": "OrderCancelledEvent", "event_source": "binance"}
2021-06-20 16:36:16,593 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Successfully cancelled order x-XEKWYICX-SFTBC1624178116034453.
2021-06-20 16:36:16,593 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178176.0, "order_id": "x-XEKWYICX-SFTBC1624178116034453", "exchange_order_id": null, "event_name": "OrderCancelledEvent", "event_source": "binance"}

# the bot now created hanging and active order
2021-06-20 16:36:17,005 - 17028 - hummingbot.strategy.hanging_orders_tracker - INFO - Updating hanging orders...
2021-06-20 16:36:17,005 - 17028 - hummingbot.strategy.hanging_orders_tracker - INFO - Original hanging orders: {LimitOrder('x-XEKWYICX-BFTBC1624177815979308', 'FRONT-BTC', True, 'FRONT', 'BTC', 0.00001961, 12.80000000)}
2021-06-20 16:36:17,005 - 17028 - hummingbot.strategy.hanging_orders_tracker - INFO - Equivalent hanging orders: frozenset({HangingOrder(order_id=None, trading_pair='FRONT-BTC', is_buy=True, price=Decimal('0.00001961'), amount=Decimal('12.80000000'))})
2021-06-20 16:36:17,005 - 17028 - hummingbot.strategy.hanging_orders_tracker - INFO - Need to create: {HangingOrder(order_id=None, trading_pair='FRONT-BTC', is_buy=True, price=Decimal('0.00001961'), amount=Decimal('12.80000000'))}
2021-06-20 16:36:17,005 - 17028 - hummingbot.strategy.hanging_orders_tracker - INFO - Need to cancel: set()
2021-06-20 16:36:17,006 - 17028 - hummingbot.strategy.avellaneda_market_making.avellaneda_market_making - INFO - (FRONT-BTC) Creating 1 bid orders at (Size, Price): ['16 FRONT, 0.00001964 BTC']
2021-06-20 16:36:17,006 - 17028 - hummingbot.strategy.avellaneda_market_making.avellaneda_market_making - INFO - (FRONT-BTC) Creating 1 ask orders at (Size, Price): ['15.9 FRONT, 0.00001967 BTC']
2021-06-20 16:36:17,104 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Created LIMIT_MAKER BUY order x-XEKWYICX-BFTBC1624178177005954 for 12.80000000 FRONT-BTC.
2021-06-20 16:36:17,105 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178177.0, "type": "OrderType.LIMIT_MAKER", "trading_pair": "FRONT-BTC", "amount": "12.80000000", "price": "0.00001961", "order_id": "x-XEKWYICX-BFTBC1624178177005954", "exchange_order_id": "19328860", "leverage": 1, "position": "NILL", "event_name": "BuyOrderCreatedEvent", "event_source": "binance"}
2021-06-20 16:36:17,608 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Created LIMIT_MAKER BUY order x-XEKWYICX-BFTBC1624178177006958 for 16.00000000 FRONT-BTC.
2021-06-20 16:36:17,608 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178177.0, "type": "OrderType.LIMIT_MAKER", "trading_pair": "FRONT-BTC", "amount": "16.00000000", "price": "0.00001964", "order_id": "x-XEKWYICX-BFTBC1624178177006958", "exchange_order_id": "19328858", "leverage": 1, "position": "NILL", "event_name": "BuyOrderCreatedEvent", "event_source": "binance"}
2021-06-20 16:36:18,112 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Created LIMIT_MAKER SELL order x-XEKWYICX-SFTBC1624178177006959 for 15.90000000 FRONT-BTC.
2021-06-20 16:36:18,112 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178178.0, "type": "OrderType.LIMIT_MAKER", "trading_pair": "FRONT-BTC", "amount": "15.90000000", "price": "0.00001967", "order_id": "x-XEKWYICX-SFTBC1624178177006959", "exchange_order_id": "19328859", "leverage": 1, "position": "NILL", "event_name": "SellOrderCreatedEvent", "event_source": "binance"}
```

Steps to reproduce:
1. Run an Avellaneda-mm with hanging order 
2. Set max_order_age to 2:0x minutes (I set mine to 187s) 
3. Once it reached the set max_order_age, wait for the hanging order to be recreated

Release version:
dev-0.41.0

Attachment:
[logs_ave-hanging-orders.log](https://github.com/CoinAlpha/hummingbot/files/6681962/logs_ave-hanging-orders.log)

**Comment by phbrgnomo** [comment]
To illustrate a bit more:

![Captura de tela 2021-06-20 00 15 06](https://user-images.githubusercontent.com/34484531/122767939-8473f600-d279-11eb-85e6-048d3e292e81.png)

As can be seen above, there was 1 hanging buy order, with a total amount of 1.7,. Right after that `status`, this sell order was taken:


![Captura de tela 2021-06-20 00 14 47](https://user-images.githubusercontent.com/34484531/122768990-87bbb180-d27a-11eb-9d73-6494ec86ac40.png)


But after checking +- 20 mins, I was expecting the total hanging order amount to be 1.7 + 0.91 = 2.61



But the last executed order was totally ignored:

![Captura de tela 2021-06-20 00 15 15](https://user-images.githubusercontent.com/34484531/122768627-2b589200-d27a-11eb-9f22-3cf17692db0a.png)

**Comment by keithbaum** [comment]
> Adding another instance of this issue on hanging orders and max_order_age:
> While looking into it, I noticed that the hanging order will only be recreated after the next order cycle. In my example, I set max_order_age to 3:07s (187) and waited for the even to recreate the hanging order. It was not recreated and the bot waited for my order_refresh_time (60s) then it creates both active and hanging orders again.
> 
>  front-btc.mp4 
> 
> ```
> # the bot created bid/ask order
> 2021-06-20 16:35:16,135 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Created LIMIT_MAKER BUY order x-XEKWYICX-BFTBC1624178116034452 for 16.00000000 FRONT-BTC.
> 2021-06-20 16:35:16,135 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178116.0, "type": "OrderType.LIMIT_MAKER", "trading_pair": "FRONT-BTC", "amount": "16.00000000", "price": "0.00001963", "order_id": "x-XEKWYICX-BFTBC1624178116034452", "exchange_order_id": "19328846", "leverage": 1, "position": "NILL", "event_name": "BuyOrderCreatedEvent", "event_source": "binance"}
> 2021-06-20 16:35:16,645 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Created LIMIT_MAKER SELL order x-XEKWYICX-SFTBC1624178116034453 for 15.90000000 FRONT-BTC.
> 2021-06-20 16:35:16,645 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178116.0, "type": "OrderType.LIMIT_MAKER", "trading_pair": "FRONT-BTC", "amount": "15.90000000", "price": "0.00001968", "order_id": "x-XEKWYICX-SFTBC1624178116034453", "exchange_order_id": "19328845", "leverage": 1, "position": "NILL", "event_name": "SellOrderCreatedEvent", "event_source": "binance"}
> 
> # the hanging order reached max_order_age and cancelled
> 2021-06-20 16:35:20,028 - 17028 - hummingbot.strategy.hanging_orders_tracker - INFO - Reached max_order_age=187.0sec hanging order: HangingOrder(order_id='x-XEKWYICX-BFTBC1624177932997306', trading_pair='FRONT-BTC', is_buy=True, price=Decimal('0.00001961'), amount=Decimal('12.80000000')). Renewing...
> 2021-06-20 16:35:20,028 - 17028 - hummingbot.strategy.avellaneda_market_making.avellaneda_market_making - INFO - (FRONT-BTC) Cancelling the limit order x-XEKWYICX-BFTBC1624177932997306. [clock=2021-06-20 08:35:20+00:00]
> 2021-06-20 16:35:20,176 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Successfully cancelled order x-XEKWYICX-BFTBC1624177932997306.
> 2021-06-20 16:35:20,176 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178120.0, "order_id": "x-XEKWYICX-BFTBC1624177932997306", "exchange_order_id": null, "event_name": "OrderCancelledEvent", "event_source": "binance"}
> 2021-06-20 16:36:15,997 - 17028 - hummingbot.strategy.avellaneda_market_making.avellaneda_market_making - INFO - (FRONT-BTC) Cancelling the limit order x-XEKWYICX-BFTBC1624178116034452. [clock=2021-06-20 08:36:16+00:00]
> 2021-06-20 16:36:15,997 - 17028 - hummingbot.strategy.avellaneda_market_making.avellaneda_market_making - INFO - (FRONT-BTC) Cancelling the limit order x-XEKWYICX-SFTBC1624178116034453. [clock=2021-06-20 08:36:16+00:00]
> 2021-06-20 16:36:16,091 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Successfully cancelled order x-XEKWYICX-BFTBC1624178116034452.
> 2021-06-20 16:36:16,091 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178176.0, "order_id": "x-XEKWYICX-BFTBC1624178116034452", "exchange_order_id": null, "event_name": "OrderCancelledEvent", "event_source": "binance"}
> 2021-06-20 16:36:16,593 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Successfully cancelled order x-XEKWYICX-SFTBC1624178116034453.
> 2021-06-20 16:36:16,593 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178176.0, "order_id": "x-XEKWYICX-SFTBC1624178116034453", "exchange_order_id": null, "event_name": "OrderCancelledEvent", "event_source": "binance"}
> 
> # the bot now created hanging and active order
> 2021-06-20 16:36:17,005 - 17028 - hummingbot.strategy.hanging_orders_tracker - INFO - Updating hanging orders...
> 2021-06-20 16:36:17,005 - 17028 - hummingbot.strategy.hanging_orders_tracker - INFO - Original hanging orders: {LimitOrder('x-XEKWYICX-BFTBC1624177815979308', 'FRONT-BTC', True, 'FRONT', 'BTC', 0.00001961, 12.80000000)}
> 2021-06-20 16:36:17,005 - 17028 - hummingbot.strategy.hanging_orders_tracker - INFO - Equivalent hanging orders: frozenset({HangingOrder(order_id=None, trading_pair='FRONT-BTC', is_buy=True, price=Decimal('0.00001961'), amount=Decimal('12.80000000'))})
> 2021-06-20 16:36:17,005 - 17028 - hummingbot.strategy.hanging_orders_tracker - INFO - Need to create: {HangingOrder(order_id=None, trading_pair='FRONT-BTC', is_buy=True, price=Decimal('0.00001961'), amount=Decimal('12.80000000'))}
> 2021-06-20 16:36:17,005 - 17028 - hummingbot.strategy.hanging_orders_tracker - INFO - Need to cancel: set()
> 2021-06-20 16:36:17,006 - 17028 - hummingbot.strategy.avellaneda_market_making.avellaneda_market_making - INFO - (FRONT-BTC) Creating 1 bid orders at (Size, Price): ['16 FRONT, 0.00001964 BTC']
> 2021-06-20 16:36:17,006 - 17028 - hummingbot.strategy.avellaneda_market_making.avellaneda_market_making - INFO - (FRONT-BTC) Creating 1 ask orders at (Size, Price): ['15.9 FRONT, 0.00001967 BTC']
> 2021-06-20 16:36:17,104 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Created LIMIT_MAKER BUY order x-XEKWYICX-BFTBC1624178177005954 for 12.80000000 FRONT-BTC.
> 2021-06-20 16:36:17,105 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178177.0, "type": "OrderType.LIMIT_MAKER", "trading_pair": "FRONT-BTC", "amount": "12.80000000", "price": "0.00001961", "order_id": "x-XEKWYICX-BFTBC1624178177005954", "exchange_order_id": "19328860", "leverage": 1, "position": "NILL", "event_name": "BuyOrderCreatedEvent", "event_source": "binance"}
> 2021-06-20 16:36:17,608 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Created LIMIT_MAKER BUY order x-XEKWYICX-BFTBC1624178177006958 for 16.00000000 FRONT-BTC.
> 2021-06-20 16:36:17,608 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178177.0, "type": "OrderType.LIMIT_MAKER", "trading_pair": "FRONT-BTC", "amount": "16.00000000", "price": "0.00001964", "order_id": "x-XEKWYICX-BFTBC1624178177006958", "exchange_order_id": "19328858", "leverage": 1, "position": "NILL", "event_name": "BuyOrderCreatedEvent", "event_source": "binance"}
> 2021-06-20 16:36:18,112 - 17028 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Created LIMIT_MAKER SELL order x-XEKWYICX-SFTBC1624178177006959 for 15.90000000 FRONT-BTC.
> 2021-06-20 16:36:18,112 - 17028 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1624178178.0, "type": "OrderType.LIMIT_MAKER", "trading_pair": "FRONT-BTC", "amount": "15.90000000", "price": "0.00001967", "order_id": "x-XEKWYICX-SFTBC1624178177006959", "exchange_order_id": "19328859", "leverage": 1, "position": "NILL", "event_name": "SellOrderCreatedEvent", "event_source": "binance"}
> ```
> 
> Steps to reproduce:
> 
> 1. Run an Avellaneda-mm with hanging order
> 2. Set max_order_age to 2:0x minutes (I set mine to 187s)
> 3. Once it reached the set max_order_age, wait for the hanging order to be recreated
> 
> Release version:
> dev-0.41.0
> 
> Attachment:
> [logs_ave-hanging-orders.log](https://github.com/CoinAlpha/hummingbot/files/6681962/logs_ave-hanging-orders.log)

Hi @rapcmia / @CrimsonJacket ,
Logic is exactly as you mentioned. Yes it will cancel the max_order_aged hanging order, and it will recreate them in the next cycle of orders. This implies there's going to be a gap in time between order cancellation and recreation but that'll be no more than 1 whole period of order_refresh_time (if max_order_age is multiple of order_refresh_time). I didn't consider that a major issue at the moment of committing the code, but it could be fixed.

**Comment by keithbaum** [comment]
> To illustrate a bit more:
> 
> ![Captura de tela 2021-06-20 00 15 06](https://user-images.githubusercontent.com/34484531/122767939-8473f600-d279-11eb-85e6-048d3e292e81.png)
> 
> As can be seen above, there was 1 hanging buy order, with a total amount of 1.7,. Right after that `status`, this sell order was taken:
> 
> ![Captura de tela 2021-06-20 00 14 47](https://user-images.githubusercontent.com/34484531/122768990-87bbb180-d27a-11eb-9d73-6494ec86ac40.png)
> 
> But after checking +- 20 mins, I was expecting the total hanging order amount to be 1.7 + 0.91 = 2.61
> 
> But the last executed order was totally ignored:
> 
> ![Captura de tela 2021-06-20 00 15 15](https://user-images.githubusercontent.com/34484531/122768627-2b589200-d27a-11eb-9f22-3cf17692db0a.png)

Hey @phbrgnomo ,
Do you happen to have the logs for this? In that way I could follow the timeline of events of the order. A priori, this might not be related with max_order_age issue mentioned before. thanks

**Comment by dennisocana** [comment]
Fixed in version 0.41 release
